### ISSUE TITLE
Out-dated documentation for dataloader?

### ISSUE THREAD

**Issue by Moelf** [FluxML/Flux.jl#1310]
per https://fluxml.ai/Flux.jl/v0.11/data/dataloader/, we see that it's possible to have the following working:
```
Xs = rand(4,1000)
Ys = rand(1000)
train_loader = Flux.Data.DataLoader((Xs, Ys), batchsize=2, shuffle=false)
model = Chain(Dense(4,1), softmax)
opt = ADAM()
ps = Flux.params(model)
Flux.train!(Flux.Losses.mse, ps, train_loader, opt)
```

but now it gives:
```
DimensionMismatch("arrays could not be broadcast to a common size; got a dimension with lengths 4 and 2")
```

**Comment by Moelf** [comment]
apparently I need to `transpose(Ys)` making it a `1x1000` matrix for this to work

**Comment by CarloLucibello** [comment]
I think that the DataLoader here is doing what is expected to do:
```julia
julia> x, y = first(train_loader);

julia> size(x)
(4, 2)

julia> size(y)
(2,)
```
Then you have to construct a loss which handles the mini-batch properly

**Comment by CarloLucibello** [comment]
btw, the current docstring doesn't contain the example you mentioned, so we should be fine
